jl4-lsp/src/LSP/Core/IdeConfiguration.hs

```diff
@@ -86,12 +86,17 @@ isWorkspaceFile uri =
       if isRelative (fromNormalizedFilePath file)
         then return True
         else do
-          IdeConfiguration {..} <- getIdeConfiguration
+          mconfig <- try getIdeConfiguration
+
           let toText = getUri . fromNormalizedUri
-          return $
-            any
-              (\root -> toText root `isPrefixOf` toText (normalizedFilePathToUri file))
-              workspaceFolders
+          pure case mconfig of
+            -- NOTE: if we are not in an editor context, we assume that every file is part
+            -- of our "workspace" as in this case there's not really a concept of a workspace
+            Left (_ :: SomeException) -> True
+            Right config ->
+              any
+                (\root -> toText root `isPrefixOf` toText (normalizedFilePathToUri file))
```

Comment:
I know you didn't write this, but I am pretty sure this check is wrong.

Reply:
I also have a weird feeling about it but I don't immediately see what's wrong about it (uris should be normalized, they should both be file uris)

Reply:
The `toText` strips the trailing slash, so this check:

```haskell
> toText = getUri . fromNormalizedUri
> toNUri = toNormalizedUri . filePathToUri
> toText (toNUri "/home/hugin/Documents/haskell/cclaw/l4-ide/jl4-lsp/src/LSP/Core/") `Text.isPrefixOf` toText (toNUri "/home/hugin/Documents/haskell/cclaw/l4-ide/jl4-lsp/src/LSP/Core-Not-Dir.hs")
True
```
This should be `False` but returns `True`.